dkg/src/inputs.rs

```diff
@@ -0,0 +1,131 @@
+use frost::{
+    keys::dkg::{round1, round2},
+    Error, Identifier,
+};
+
+use frost_ed25519 as frost;
+
+use eyre::eyre;
+
+use std::io::{BufRead, Write};
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct Config {
+    pub min_signers: u16,
+    pub max_signers: u16,
+    pub identifier: Identifier,
+}
+
+pub trait Logger {
+    fn log(&mut self, value: String);
+}
+
+fn validate_inputs(config: &Config) -> Result<(), Error> {
+    if config.min_signers < 2 {
+        return Err(Error::InvalidMinSigners);
+    }
+
+    if config.max_signers < 2 {
+        return Err(Error::InvalidMaxSigners);
+    }
+
+    if config.min_signers > config.max_signers {
+        return Err(Error::InvalidMinSigners);
+    }
+
+    Ok(())
+}
+
+pub fn request_inputs(
+    input: &mut impl BufRead,
+    logger: &mut dyn Write,
+) -> Result<Config, Box<dyn std::error::Error>> {
+    writeln!(logger, "The minimum number of signers: (2 or more)")?;
+
+    let mut min = String::new();
+    input.read_line(&mut min)?;
+
+    let min_signers = min
+        .trim()
+        .parse::<u16>()
+        .map_err(|_| eyre!("Invalid minimum number of signers"))?;
+
+    writeln!(logger, "The maximum number of signers: ")?;
+
+    let mut max = String::new();
+    input.read_line(&mut max)?;
+    let max_signers = max
+        .trim()
+        .parse::<u16>()
+        .map_err(|_| eyre!("Invalid maximum number of signers"))?;
+
+    writeln!(
+        logger,
+        "Your identifier (this should be an integer between 1 and 65535):"
+    )?;
+
+    let mut identifier_input = String::new();
+
+    input.read_line(&mut identifier_input)?;
+
+    let u16_identifier = identifier_input
+        .trim()
+        .parse::<u16>()
+        .map_err(|_| eyre!("Invalid identifier"))?;
+    let identifier = u16_identifier.try_into()?;
+
+    let config = Config {
+        min_signers,
+        max_signers,
+        identifier,
+    };
+
+    validate_inputs(&config)?;
+
+    Ok(config)
+}
+
+pub fn read_identifier(input: &mut impl BufRead) -> Result<Identifier, Box<dyn std::error::Error>> {
```

Comment:
I couldn't add an entry here without getting an error. It seems the identifier_input is being requested. The identifier_input on line 103 isn't ever used. It's unclear to me what input you're trying to get here.

Reply:
I should have learned at this point to not do any refactorings right before opening the PR without proper testing :see_no_evil: Sorry about that. Fixed in 8311efe9ff96f448c8384a059f6d1cee56c8472f